src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using ManifestReaderTests;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.DotNet.Workloads.Workload.Install;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using NuGet.Versioning;
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using System.Text.Json;
+using Microsoft.TemplateEngine.Edge.Constraints;
+
+namespace Microsoft.DotNet.Cli.Workload.Install.Tests
+{
+    public class WorkloadGarbageCollectionTests : SdkTest
+    {
+        private readonly BufferedReporter _reporter;
+        private string _testDirectory;
+        private string _dotnetRoot;
+
+        public WorkloadGarbageCollectionTests(ITestOutputHelper log) : base(log)
+        {
+            _reporter = new BufferedReporter();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanGarbageCollect()
+        {
+            CreateMockManifest("testmanifest", "1.0.0", "6.0.100", sourceManifestName: @"Sample2.json");
+            CreateMockManifest("testmanifest", "2.0.0", "6.0.300", sourceManifestName: @"Sample2_v2.json");
+
+            var (installer, getResolver) = GetTestInstaller();
+            var packsToKeep = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk),
+                CreatePackInfo("Xamarin.Android.Framework", "8.5.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.5.0.1", WorkloadPackKind.Library)
+            };
+
+            var packsToCollect = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Framework", "8.4.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.4.7.4", WorkloadPackKind.Library)
+            };
+            var sdkVersions = new [] { "6.0.100", "6.0.300" };
+
+            // Write packs
+            foreach (var sdkVersion in sdkVersions)
+            {
+                foreach (var pack in packsToKeep.Concat(packsToCollect))
+                {
+                    CreateInstalledPack(pack, sdkVersion);
```

Comment:
I see you have cleaned up the tests quite a bit, I quite like this `CreateInstalledPack` function to de-duplicate the logic. :smile: